osu.Game/Rulesets/Objects/HitObject.cs

```diff
@@ -155,6 +136,20 @@ public void ApplyDefaults(ControlPointInfo controlPointInfo, IBeatmapDifficultyI
             foreach (var h in nestedHitObjects)
                 h.ApplyDefaults(controlPointInfo, difficulty, cancellationToken);
 
+            // importantly, this callback is only registered after default application
+            // to ensure that the read of `this.GetEndTime()` within doesn't return an invalid value
+            // if `StartTimeBindable` is changed prior to default application.
+            StartTimeBindable.ValueChanged += time =>
```

Comment:
ApplyDefaults can be called multiple times, so this needs to be unbound as well.

Reply:
Not sure what the best place to unbind would be - the very start of `ApplyDefaults()`, perhaps?

Reply:
Or just before the bind.